### PR TITLE
Fix tests and lint

### PR DESCRIPTION
**Pull request by nwalters512**
This fixes the failing test called out in https://github.com/dividab/tsconfig-paths/pull/174#discussion_r863079371. It also fixes a few more failing tests. The entire test suite now passes!

I also took the opportunity to remove some commented-out assertions and remove a few unnecessary bits that had `// eslint-disable...` comments.

I would strongly encourage the maintainers to set up CI to run on pull requests, not just on master, so that errors can be caught quickly and resolved before changes are merged!

### CODE REVIEW

File: src/__tests__/config-loader.test.ts [hunk 1]
```diff
@@ -49,38 +44,31 @@ describe("config-loader", (): void => {
...
 
     const successResult = result as ConfigLoaderSuccessResult;
-    // assert.equal(successResult.resultType, "success");
-    // assert.equal(successResult.absoluteBaseUrl, join("/baz", "src"));
     expect(successResult.resultType).toBe("success");
     expect(successResult.absoluteBaseUrl).toBe(join("/baz", "src"));
   });
 
-  it("should show an error message when baseUrl is missing", () => {
+  it("should tolerate a missing baseUrl", () => {
```
Comment (nwalters512) [inline src/__tests__/config-loader.test.ts:59]: I believe this test was broken by https://github.com/dividab/tsconfig-paths/pull/208.

File: src/__tests__/config-loader.test.ts [hunk 2]
```diff
@@ -102,15 +88,14 @@ describe("config-loader", (): void => {
...
+      tsConfigLoader: () => ({
         tsConfigPath: "/baz/tsconfig.json",
         baseUrl: "/baz",
         paths: {},
       }),
     });
 
     const successResult = result as ConfigLoaderSuccessResult;
-    assert.equal(successResult.absoluteBaseUrl, "/baz");
+    expect(successResult.absoluteBaseUrl).toEqual("/baz");
```
Comment (nwalters512) [inline src/__tests__/config-loader.test.ts:99]: This was the failing test from https://github.com/dividab/tsconfig-paths/pull/174.

File: src/__tests__/mapping-entry.test.ts [hunk 3]
```diff
@@ -6,56 +6,30 @@ describe("mapping-entry", () => {
...
-    // ]);
     expect(result).toEqual([
       {
         pattern: "longest/pre/fix/*",
         paths: [join("/absolute", "base", "url", "foo2", "bar")],
       },
       {
         pattern: "pre/fix/*",
-        paths: [join("/absolute", "base", "url", "foo3")],
+        paths: [join("/foo3")],
```
Comment (nwalters512) [inline src/__tests__/mapping-entry.test.ts:22]: I believe these test cases were broken by https://github.com/dividab/tsconfig-paths/pull/184. I updated them to have a sampling of both relative and absolute paths.

**Comment by jonaskello** [comment]
@nwalters512 I was trying to get the tests running in CI for the PR. Could you try to merge master into this and push again?

**Comment by nwalters512** [comment]
@jonaskello I believe you're missing configuration from `.github/workflows/ci.yml`. You need `on: [push, pull_request]` instead of just `on: [push]`. The `push` event only handles pushes to this repo, not pushes to forks, which will be the workflow for external contributors. See https://github.com/scinos/yarn-deduplicate/blob/1d0906c46895ffc4a2f22ef762ed6e8474327944/.github/workflows/node.js.yml#L6 for an example.

**Comment by jonaskello** [comment]
@nwalters512 Ah, thanks, that may be it! It was running for the PR where I changed the workflow but I guess counts as just push. I updated master according to your example, could you try to merge it into this PR again?

**Comment by nwalters512** [comment]
@jonaskello looks like we're good now!

**Comment by jonaskello** [comment]
@nwalters512 Perfect, thanks for the hint :-)

**Pull request merged**